src/_zkapauthorizer/tests/matchers.py

```diff
@@ -50,13 +52,22 @@
 )
 from testtools.twistedsupport import succeeded
 from treq import content
+from twisted.web.http_headers import Headers
 
 from ..model import Pass
 from ..server.spending import _SpendingData
 from ._exception import raises
 from ._float_matchers import matches_float_within_distance
 from ._sql_matchers import equals_database
 
+_T = TypeVar("_T")
+
+
+class Matcher(_Matcher, Generic[_T]):
+    """
+    A generic version of ``_Matcher``.
```

Comment:
in other words, a type that lets you write `Matcher[foo]` to say "this is a thing that matches on instances of foo".